components/forms/CreateRoleForm.tsx

```diff
@@ -1,45 +1,64 @@
-import { Form, InputNumber, Input, Button } from 'antd';
+import { Form, InputNumber, Input, Modal } from 'antd';
 import rolesApi from '../../api/rolesApi';
-import { RolePostData } from '../../types/role';
+import { CompanyListData } from '../../types/company';
+import { RoleData } from '../../types/role';
+import { Nullable } from '../../types/utils';
+import { DEFAULT_NUM_MONTHS_BEFORE_ROLE_START } from '../../utils/constants';
+import CompanyOption from '../CompanyOption';
 import RoleTypesSelect from './RoleTypesSelect';
 
-function CreateRoleForm() {
+type Props = {
+  company: Nullable<CompanyListData>;
+
+  isOpen: boolean;
+  closeForm: () => void;
+  onCreate: (role: RoleData) => void;
+};
+
+function getDefaultRoleYear(): number {
+  const date = new Date();
+  date.setMonth(date.getMonth() + DEFAULT_NUM_MONTHS_BEFORE_ROLE_START);
+  return date.getFullYear();
+}
+
+function CreateRoleForm({ company, isOpen, closeForm, onCreate }: Props) {
```

Comment:
There's a bug with this modal.

https://user-images.githubusercontent.com/53614454/190858841-aa30df30-ed54-4cb8-9740-4bfd3ed57f61.mp4

When the company is empty, the modal for the role cannot be opened.

Reply:
Right, this is something that I wanted to discuss. Should the role field be disabled when the company is not filled? 

Not sure if this is the best UX though.

Reply:
> Right, this is something that I wanted to discuss. Should the role field be disabled when the company is not filled?
> 
> Not sure if this is the best UX though.

I think we can disable it and show a message like how the validation message are shown.